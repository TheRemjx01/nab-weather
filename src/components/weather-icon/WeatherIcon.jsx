import React from 'react';

const iconMap = {
 sn: { link: 'https://www.metaweather.com/static/img/weather/sn.svg', name:'Snow' },
 sl: { link: 'https://www.metaweather.com/static/img/weather/sl.svg', name: 'Sleet' },
 h: {link: 'https://www.metaweather.com/static/img/weather/h.svg', name: 'Hail' },
 t: {link: 'https://www.metaweather.com/static/img/weather/t.svg', name:'Thunderstorm'},
 hr: {link: 'https://www.metaweather.com/static/img/weather/hr.svg', name: 'Heavy Rain'},
 lr: {link: 'https://www.metaweather.com/static/img/weather/lr.svg', name: 'Light Rain'},
 s: {link: 'https://www.metaweather.com/static/img/weather/s.svg', name: 'Showers'},
 hc: {link: 'https://www.metaweather.com/static/img/weather/hc.svg', name: 'Heavy Cloud' },
 lc: {link: 'https://www.metaweather.com/static/img/weather/lc.svg', name: 'Light Cloud' },
 c: {link: 'https://www.metaweather.com/static/img/weather/c.svg', name: 'Clear'}
}

const WeatherIcon = ({weather}) => {
 return (
  <img src={iconMap[weather].link} alt={iconMap[weather].name} style={{width: '50px', height: '50px'}}/>
 );
};

export default WeatherIcon;