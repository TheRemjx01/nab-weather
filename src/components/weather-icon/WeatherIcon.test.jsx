import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WeatherIcon from "./WeatherIcon";

describe('<WeatherIcon />', () => {
 it('should be able to render', () => {
  const weather = 'sn';
  const className='mock-class'
  const Wrapper = shallow(<WeatherIcon weather={weather} className={className}/>);
  expect(toJson(Wrapper)).toMatchSnapshot();
 })
})