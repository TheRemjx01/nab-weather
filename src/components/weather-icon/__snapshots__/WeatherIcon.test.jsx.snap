// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherIcon /> should be able to render 1`] = `
<img
  alt="Snow"
  className="mock-class"
  src="https://www.metaweather.com/static/img/weather/sn.svg"
/>
`;
